### ISSUE TITLE
Input mapping returning to default when controller changes

### ISSUE THREAD

**Issue by Wellvergton** [libretro/retroarch-snap#23]
## Description

I have a DualShock 3 and recently bought Retroflag's classic Mega Drive controller, I made the mapping of the buttons and everything went well until I went back to using DualShock and the mapping for it got lost, I found it strange, but I did the mapping again, however, when I was playing again with the Retroflag's controller the same problem happened, the mapping I did previously was lost.

If I just exit the Retroarch to return afterwards to play with the same controller I used the last time the mapping is maintained.

### Expected behavior

Keep the input mapping

### Actual behavior

If the device index changes the mapping are lost

### Steps to reproduce the bug

1. Map a controller
2. Change the controller to one of another brand
3. Map this controller
4. Return to the previous controller

### Version/Commit

- RetroArch: 1.7.6/7ffeebf

### Environment information

- OS: Ubuntu 16.04.5

**Comment by hizzlekizzle** [comment]
Are you creating autoconfigs for the pads?

**Comment by Wellvergton** [comment]
Yes. The strange thing is that the autoconfig file doesn't update when the config are lost, it only updates when I remap again but nothing changes in the files.

**Comment by fr500** [comment]
Probably the files are in a read only location then

**Comment by Wellvergton** [comment]
I think not. I think I did not express myself well. When I manually create autoconfig the file updates normally, which shows that it is not read only, what I saying is that when the settings are lost and I will check the file it still has the remap information that I entered, but in the Retroarch the commands do not match the autoconfig file for the controller. And if I do the mapping again the file will be updated but the information will be the same as the last mapping.
